server.js

```diff
@@ -572,6 +574,111 @@ cache(function(data, match, sendBadge, request) {
+  var branch = match[3];
+  var format = match[4];
+  var apiUrl = gitlabApiUrl + '/projects/' + encodeURIComponent(userRepo) + '/pipelines';
+  console.log(apiUrl);
+  var badgeData = getBadgeData({'s':'build', 'c': 'coverage'}[dataKind], data);
+  var callback = function(branch, page){
+    var qs = {
+      page: page,
+      per_page: 100,
+    };
+    gitlabAuth.request(request, apiUrl, qs, function(err, res, buffer) {
+      if (err != null) {
+        badgeData.text[1] = 'inaccessible';
+        sendBadge(format, badgeData);
+        return;
+      }
+      try {
+        var data = JSON.parse(buffer);
+        var done = false;
+        // Assume sorted by time
+        if(page >= 10 || data.length == 0) throw 'No match';
+        for(var i = 0; i < data.length; ++i) {
+          if(data[i].ref === branch && ['success', 'failed'].indexOf(data[i].status) != -1) {
+            switch(dataKind) {
+              case 's':
+              var state = data[i].status;
+              badgeData.text[1] = state;
+              if (state === 'success') {
+                badgeData.colorscheme = 'brightgreen';
+              } else if (state === 'failed') {
+                badgeData.colorscheme = 'red';
+              }
+              sendBadge(format, badgeData);
+              break;
+              case 'c':
+              var buildApiUrl = gitlabApiUrl + '/projects/' + encodeURIComponent(userRepo) +
+              '/repository/commits/' + data[i].sha + '/builds';
+              gitlabAuth.request(request, buildApiUrl, {per_page: 100}, function(err, res, buffer) {
+                if (err != null) {
+                  badgeData.text[1] = 'inaccessible';
+                  sendBadge(format, badgeData);
+                  return;
+                }
+                try {
+                  var data = JSON.parse(buffer);
+                  var done = false;
+                  for(var i = 0; i < data.length; ++i) {
+                    if(data[i].coverage != null) {
+                      console.log(data[i].coverage)
+                      var percentage = parseFloat(data[i].coverage);
+                      if(percentage === NaN) throw 'NaN';
+                      badgeData.text[1] = percentage.toFixed(0) + '%';
+                      badgeData.colorscheme = coveragePercentageColor(percentage);
+                      done = true;
+                    }
+                  }
+                  if(!done) throw 'No coverage match'
+                } catch(e) {
+                  badgeData.text[1] = 'unknown';
+                }
+                sendBadge(format, badgeData);
+              });
+              break;
+            }
+            done = true;
+            break;
+          }
+        }
+        if(!done) {
+          callback(branch, page+1);
+        }
+      } catch(e) {
+        badgeData.text[1] = 'unknown';
+        sendBadge(format, badgeData);
+      }
+    });
+  }
+  if(branch == null) {
```

Comment:
To reduce complexity, could the branch default to `master` unless the user specifies a different one in the badge? Alternatively, the user could be required to specify a branch name.